### DISCUSSION TITLE
Preserve exported values immediately upon declaration like "export const varA = 42;"

### DISCUSSION THREAD

**Discussion by olivierpascal** [rollup/rollup#5339] [Q&A]
When building to esm, how to preserve exported value immediately upon declaration like `export const varA = 42;`, and to NOT systematically group exports to `const varA = 42; export { varA };`?

**Comment by lukastaegert** [comment]
This is not possible at the moment and would be rather difficult to implement on Rollup side. It is also problematic as variables need to be renamed quite often, and once that happens, you need to write the export separate anyway. The size advantages would also be negligible, especially if the output is also minified, and there is also no tangible performance advantage.

**Reply by olivierpascal** [reply]
Thanks for your reply. FYI this is related to https://github.com/facebook/stylex/issues/348 and this is a requirement to use their lib.
